### PR TITLE
Upgrade Web app to React 17

### PR DESCRIPTION
**Pull request by nevyangelova**
#### Summary
This PR addresses the React 17 upgrade, by fixing as many peer dependency errors as possible, however some libraries we use are at their latest version. We potentially need to add the `--force` flag to our install commands until we can replace them. Which is going to be owned by Web platform. 

#### Ticket Link
[MM-34187](https://mattermost.atlassian.net/browse/MM-34187)

#### Release Note
```release-note
Upgraded Mattermost web app to React 17
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,7 +44,7 @@ node_modules: package.json package-lock.json
 	node skip_integrity_check.js
 
 ifeq ($(CI),false)
-	npm install --force
```
Comment (M-ZubairAhmed) [inline Makefile]: 🥳
In my personal opinion forking repositories in mattermost is better than force.

File: .npmrc [hunk 2]
```diff
@@ -1,2 +1,2 @@
 save-exact=true
-
+legacy-peer-deps=true
```
Comment (M-ZubairAhmed) [inline .npmrc:2]: nit : extra line at the end of file

**Comment by nevyangelova** [comment]
https://mattermost-cypress-report.s3.amazonaws.com/265929-3f0691a-onprem-ent-webapp-pr-11558-mm-ee-test:3f0691a/mochawesome.html test report

**Review by hmhealey — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thanks @nevyangelova 
There is an issue on the test server with rhs keeping the draft of the message that just posted 

https://user-images.githubusercontent.com/52937121/204034328-c2f23922-114b-49d1-998f-6ff4b4a7c7d5.mov


[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/265929-3f0691a-onprem-ent-webapp-pr-11558-mm-ee-test:3f0691a/mochawesome.html) has few failures that differ from master, but are not reproducing manually. I am disregarding draw plugin ones as unrelated, but DM category test may need refactoring.

**Comment by nevyangelova** [comment]
@jgilliam17 this bug appears on master too. Unrelated to this PR. This PR just updates some libraries and is not introducing new behaviour

**Comment by nevyangelova** [comment]
@jgilliam17 this bug appears on master too. Unrelated to this PR. This PR just updates some libraries and is not introducing new behaviour

**Comment by jgilliam17** [comment]
ah, you're right Now I remember seeing this ticket filed 
Do you think that the e2es I mentioned are ok, maybe just flaky?

**Comment by nevyangelova** [comment]
i think so but we can run again @jgilliam17 just to make sure. New features keep coming in from master and introducing new bugs :/

**Comment by jgilliam17** [comment]
I'll rerun e2es, @nevyangelova can you please fix the checks? They are not all green

**Comment by nevyangelova** [comment]
all green now :)

**Comment by jgilliam17** [comment]
@nevyangelova there's an issue with cypress pipeline right now, cycle I triggered is failing to start.

**Review by jgilliam17 — APPROVED** [review]
@nevyangelova [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/266969-836daeb-onprem-ent-webapp-pr-11558-mm-ee-test:836daeb/mochawesome.html) posted and doesn't any failures that differs from [master](https://mattermost-cypress-report.s3.amazonaws.com/266994-onprem-ent-master-mm-ee-test:test/mochawesome.html)  except draw plugin ones as before.
Looks good to merge.

**Pull request merged**